src/game_controller.py

```diff
@@ -0,0 +1,99 @@
+from messenger import Messenger
+from screen import Screen
+from utils.misc import kill_thread
+
+
+class GameController:
+
+    def __init__(self):
+        self.screen = None
+        self.health_manager = None
+        self.death_manager = None
+        self.game_recovery = None
+        self.game_stats = None
+
+    def run_bot(self, config: Config, pick_corpse: bool = False):
+        keyboard.remove_hotkey(config.general['resume_key'])
+        # Create global variables if not done yet (id est: first start)
+        if not self.screen:
+            self.screen = Screen(config.general["monitor"])
+        if not self.health_manager:
+            # Run health monitor thread
+            self.health_manager = HealthManager(self.screen)
+            health_monitor_thread = threading.Thread(target=self.health_manager.start_monitor)
+            health_monitor_thread.daemon = True
+            health_monitor_thread.start()
+        if not self.death_manager:
+            # Run death monitor thread
+            self.death_manager = DeathManager(self.screen)
+            death_monitor_thread = threading.Thread(target=self.death_manager.start_monitor)
+            death_monitor_thread.daemon = True
+            death_monitor_thread.start()
+        if not self.game_recovery:
+            self.game_recovery = GameRecovery(self.screen, self.death_manager)
+        if not self.game_stats:
+            self.game_stats = GameStats()
+        # Start bot thread
+        bot = Bot(self.screen, self.game_stats, pick_corpse)
+        bot_thread = threading.Thread(target=bot.start)
+        bot_thread.daemon = True
+        bot_thread.start()
+        # Register that thread to the death and health manager so they can stop the bot thread if needed
+        self.death_manager.set_callback(lambda: bot.stop() or kill_thread(bot_thread))
+        self.health_manager.set_callback(lambda: bot.stop() or kill_thread(bot_thread))
+        self.health_manager.set_belt_manager(bot.get_belt_manager())
+        do_restart = False
+        keyboard.add_hotkey(config.general['resume_key'], lambda: bot.toggle_pause())
+        messenger = Messenger()
+        while 1:
+            self.health_manager.update_location(bot.get_curr_location())
+            max_game_length_reached = self.game_stats.get_current_game_length() > config.general["max_game_length_s"]
+            if max_game_length_reached or self.death_manager.died() or self.health_manager.did_chicken():
+                # Some debug and logging
+                if max_game_length_reached:
+                    Logger.info(f"Max game length reached. Attempting to restart {config.general['name']}!")
+                    if config.general["info_screenshots"]:
+                        cv2.imwrite(
+                            "./info_screenshots/info_max_game_length_reached_" + time.strftime(
+                                "%Y%m%d_%H%M%S") + ".png",
+                            bot._screen.grab())
```

Comment:
I understand that some ppl have ocd for line length. But this could maybe be moved to L72? ;)

Reply:
Done, I've also made it access the self.screen rather than bot screen as it is the same.
I'm using PyCharm I set the line wrap at 180 chars instead of 120 for this project.